int32 cluster_id
float64 avg_x
float64 avg_y
float64 avg_z
